src/seedsigner/views/screensaver.py

```diff
@@ -150,28 +150,39 @@ def __init__(self, buttons):
 
         self.min_coords = (0, 0)
         self.max_coords = (self.logo.size[0], self.logo.size[1])
-
-        self.increment_x = self.rand_increment()
-        self.increment_y = self.rand_increment()
+        
+        # Movement vector components - includes both direction and magnitude
+        self.vector_x = self.rand_direction()
+        self.vector_y = self.rand_direction()
+        
         self.cur_x = int(self.logo.size[0] / 2)
         self.cur_y = int(self.logo.size[1] / 2)
 
         self._is_running = False
         self.last_screen = None
 
+        # Calculate min/max speed based on minimum screen dimension
+        min_screen_dimension = min(self.renderer.canvas_width, self.renderer.canvas_height)
+        # Min speed: cross the screen in 3 seconds, max: cross in 1 second
+        self.min_speed = min_screen_dimension / 3.0  # pixels per second
+        self.max_speed = min_screen_dimension / 1.0  # pixels per second
```

Comment:
In my testing, min / max speeds worked best if it took 1.5s - 8s to cross the screen in any one dimension.

But note that because of your `rand_direction` implementation, the speeds specified here do NOT correspond to the actual min/max speed you're achieving.

`min_speed` is X but you're scaling it by a fractional `rand_direction` so the actual minimum speed can be much slower, approaching zero.